GTBitsOfGood/umi-feeds-app
# Michael/dish bug fixes

## Dish Bug Fixes

Makes sure that the DonationList Screen is using the expected versions of dishes either DonationDish or Dish.  Also adds functionality to buttons on DonationList Screen and properly links View Donation to show the information about the correct donation that you requested information about.

### How to Test

Add a few items to the cart and look at the cart screen.  You'll see the correct names.  Adding and removing to the list works.  Changing quantity works.  Clicking to view the donation will let you see that donation item's details.  The button at the bottom properly links to the corresponding views.

### Important Changes

- Added a Mapping from ID to Dish in the DonationList Screen as a part of the state of the screen.  This map updates if the length of the user's dishes changes at all when a dish is requested.  I considered if this could introduce a bug if something were added and something else removed from the user's dishes between calls, so I also added a check if the retrieved dish was undefined to try to update the mapping.

### Related Github Issues

- resolves https://github.com/GTBitsOfGood/umi-feeds-app/issues/126

### Related PRs

- NA

### Checklist
